pkg/kubelet/pluginmanager/pluginwatcher/plugin_watcher.go

```
@@ -93,6 +94,12 @@ func (w *Watcher) Start(stopCh <-chan struct{}) error {
 		}
 	}(fsWatcher)
 
+	// Timed polling traversal plugin dir, Prevent failure to create plugin after getting file event, no subsequent events
+	go wait.Until(func() {
+		if err := w.traversePluginDir(w.path); err != nil {
```

I'm not sure how calling `traversePluginDir` every minute helps to resolve the issue? Plugin directory is already being watched at this time and existing code should react on creation and deletion of sockets. Can you elaborate?

---

The current event handling mechanism works fine under normal circumstances. For example, when the kubelet restarts and calls traversePluginDir to load plugins, if the corresponding plugin encounters a restart or any other socket-related exception, the current logic ignores this exception. As a result, there will be no subsequent events related to that socket file unless the plugin is updated or the kubelet restarts and invokes traversePluginDir again.

---

Wouldn't calling `traversePluginDir` cause re-registering already registered plugins every minute? In any case it would be good to cover this cod with e2e tests or at least modify existing one if it already exists.

---

@bart0sh Thank you, I will add the test code a bit later.